k3s-io/k3s
# etcd controller detect ghost member and update EtcdIsVoter condition

#### Proposed Changes ####

When etcd controller detects ghost member (a etcd member which don't belong to any k3s node), it set all controlplane node EtcdIsVoter condition to false with reason `UnknownMember`.

#### Types of Changes ####

Bugfix for issue https://github.com/k3s-io/k3s/issues/11231

#### Verification ####

1. create 3 controlplane node k3s cluster
2. shutdown 1 controlplane machine and delete that node by kubectl delete node
3. check node by `kubectl get node -oyaml`, the cp node EtcdIsVoter condition shall in `false` status with reason `UnknownMember`

#### Testing ####




`TODO`

#### Linked Issues ####

https://github.com/k3s-io/k3s/issues/11231

#### User-Facing Change ####
```release-note
NONE
```

#### Further Comments ####
